src/crimson/os/seastore/seastore_types.h

```diff
@@ -36,28 +36,132 @@ struct seastore_meta_t {
   }
 };
 
+// identifies a specific physical device within seastore
+using device_id_t = uint8_t;
+// order of device_id_t
+constexpr uint16_t device_id_len = 4;
+// maximum devices supported
+constexpr uint16_t max_devices = 1 << device_id_len;
+
+// segment ids without a device id encapsulated
+using device_segment_id_t = uint32_t;
+
+struct segment_id_le_t;
```

Comment:
I don't think you actually need this for the friend struct segment_id_le_t bit below.